### PR TITLE
Implementation of InputBatcher data class

### PR DESCRIPTION
**Pull request by fitzlithepius**
1. Implement a data class called BatcherInput (utils.py)
2. Change the few-shot cases relevant codes (utils.py, engine.py)
3. Modify all data classes using the new data class.
4. Add Support for segmentation mask in the batcher output.

### CODE REVIEW

**Comment by corentin-ryr** [comment]
Very good work!

I noticed the concatenation of two BatcherInputs was a bit complicated (when adding the fewshot prompt in `engine`) so maybe you could overload the add operator to simplify it. For example in `BatcherInput`: 

```python
def __add__(self, other: 'BatcherInput') -> 'BatcherInput':
    if not isinstance(other, BatcherInput):
        return NotImplemented
    # Combine the attributes of both instances
    return BatcherInput(
        conversation=self.conversation + other.conversation,
        images=(self.images or []) + (other.images or []),
        segmentation_masks=(self.segmentation_masks or []) + (other.segmentation_masks or [])
    )
```

File: multimedeval/utils.py [hunk 1]
```diff
@@ -46,18 +47,18 @@ def get_prompt(self):
...
+                single_turn_input = self.format_question(
                     self.train_dataset[index],
                     prompt=True,
                 )
-                prompt += text
-                images += img
-            self._prompt = (prompt, images)
-
+                
+                batcher_input.conversation.extend(single_turn_input.conversation)
```
Comment (corentin-ryr) [inline multimedeval/utils.py:58]: Would the "+" operator make sense here as well?

**Comment by corentin-ryr** [comment]
LGTM except for the typo in the BatcherInput class

**Review by corentin-ryr — APPROVED** [review]

File: multimedeval/utils.py [hunk 2]
```diff
@@ -209,6 +210,43 @@ class EvaluationOutput:
...
+    images: Optional[List[Image]] = field(default_factory = list)
+    segmentation_masks: Optional[List[Image]] = field(default_factory = list)
+
+    def _add_text_prompt(self, role: Literal["assistant", "user", "system"], content: str):
+        self.conversation.append({
+            "role": role,
+            "content": content
+        })
+
+    def _add_images(self, image: Image|List[Image]):
```
Comment (corentin-ryr) [inline multimedeval/utils.py:227]: The | operator is not supported in python 3.9, you should use Union[Image, List[Image]]

Comment (fitzlithepius) [reply multimedeval/utils.py:227]: Okay, I didn't know we will be sticking with Python 3.9. I will fix this in the further commit.

File: multimedeval/utils.py [hunk 3]
```diff
@@ -209,6 +210,43 @@ class EvaluationOutput:
...
+            self.images.extend(image)
+        else:
+            self.images.append(image)
+        
+    
+    def _add_segmentation_mask(self, seg_mask: Image|List[Image]):
+        if isinstance(seg_mask, list):
+            self.segmentation_masks.extend(seg_mask)
+        else:
+            self.s.append(seg_mask)
```
[the remark this answers is not in the document]
Comment (fitzlithepius) [reply multimedeval/utils.py:238]: Yes. My idea was this BatcherInput class will be partially exposed to the _batcher user_, and Java habits make me turn this into "private" to prevent the _batcher user_ directly modify what's going in the batcher.

File: multimedeval/utils.py (hunk 1 above)
Comment (fitzlithepius) [reply multimedeval/utils.py:58]: What do you think here?

Comment (corentin-ryr) [reply multimedeval/utils.py:59]: I think it will be fine, it will not create new copies of the images and the overhead of a dataclass is probably minimal.

File: multimedeval/utils.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (corentin-ryr) [reply multimedeval/utils.py:239]: Ok we can keep it like that
